### PR TITLE
feat: generate tslint.json

### PR DESCRIPTION
**Pull request by ofrobots**
`init` now generates a tslint.json file which inherits from the default
configuration. This empowers the users to customize if necessary
instead of being overly opinionated.

Fixes: #127
Fixes: #119 

This depends upon #158 and #159, so the first two commits don't need to be reviewed. They will be removed once those PRs land.

### CODE REVIEW

**Comment by JustinBeckwith** [comment]
I am 👎  on this approach.  My hope would be that the average user wouldn't need to have their own tslint.json.  I'm mostly 👎 on supporting the override at all, but I get why some people will just insist on it.  Instead of generating this during `init`, could we add a command that drops the file for you?  Like a `gts customize-lint` or something like that?

**Comment by ofrobots** [comment]
This is a good discussion.

This is what the README says about this project:
> **No configuration**. The easiest way to enforce consistent style in your project. Just drop it in.

This also speaks to what you are trying to argue about the average user not needing to customize anything. I agree with that. Yet, this project, today, generates a `tsconfig.json` in the users's directory. Is that an invitation for the user to customize it? Maybe, maybe not (IMHO I don't think it is an invitation to customize). It is just how we wire things up. At the same time, we don't want this project to be too paternalistic and _prevent_ people from customizing `tsconfig.json`. By the same argument, I do not think that generating a `tslint.json` is an invitation for the user to customize the lint config.

Either way, we need to be consistent in our own reason. We need to have a good reason why it is okay to generate `tsconfig.json` and not the `tslint.json`. We should generate both, or neither.

**Comment by JustinBeckwith** [comment]
Fair points!  I suppose in a perfect world, I'd prefer we take the same approach with `tsconfig.json`. As long as there was an easy way to generate it, not needing to have that either would be great :)

File: README.md [hunk 1]
```diff
@@ -43,8 +43,6 @@ When you run the `npx gts init` command, it's going to do a few things for you:
   - `compile`: Compiles the source code using TypeScript compiler.
   - `pretest`, `posttest` and `prepare`: convenience integrations.
 
-We strongly recommend you use the default style config, but if you must tweak, you can edit the generated `tsconfig.json`. For linter, we use the default `tslint.json` unless we find that file in your project directory.
-
```
Comment (ofrobots) [inline README.md]: Intentionally removed. If we are going to generate `tslint.json` – we do not need to document further. Let's not invite customization.

**Comment by ofrobots** [comment]
I think generating the files is friendlier. The value of this project is in providing a good, consistent style guide that can be used in a project without thinking about it.

If someone has thought about the precise, but different, style they want, and still find value in this project – let us not turn them away.

**Comment by ofrobots** [comment]
> As long as there was an easy way to generate it, not needing to have that either would be great :)

I do not think this is practical. The user *needs* to be able to customize their repo structure.

**Comment by ofrobots** [comment]
The other reason why this is desirable is mentioned in https://github.com/google/ts-style/issues/119: IDE support. A `tslint.json` in the correct location ends up getting picked up by the IDE or other tooling the user may have,.

I'm -1 on generating the configuration only under a flag, if provided by the user. That is not going to be discoverable. This needs to be the default behavior so that users don't have to pass a flag effectively equivalent to `--and-work-with-my-IDE`. I feel strong about this at this point.

**Comment by JustinBeckwith** [comment]
Again, all fair points.  You're probably right about this, but I that doesn't mean I have to be happy about it 😆 You won me over.  I'm cool with this approach.

**Pull request merged**